include/net/openthread.h

```diff
@@ -30,10 +30,16 @@ struct openthread_context {
 	struct pkt_list_elem pkt_list[CONFIG_OPENTHREAD_PKT_LIST_SIZE];
 };
 
+void openthread_set_state_changed_cb(otStateChangedCallback cb);
```

Comment:
We are missing documentation to these functions. This is a public header file so we should have documentation to relevant structs/functions in this file. If something is suppose to be private, the `openthread_context` looks like a good candidate, then please mark them using (see examples in other header files in this directory)
```
/** @cond INTERNAL_HIDDEN */
...
/* @endcond */
```